### PR TITLE
140 flesh out results page for spoken practice tool

### PR DESCRIPTION
**Pull request by dpmacisaac**

### CODE REVIEW

**Comment by dpmacisaac** [comment]
Noticed there are some issues, Ill have to figure them out before we can merge

File: ComSemApp/student/views.py [hunk 1]
```diff
@@ -8,7 +8,7 @@
 from django.contrib.auth.decorators import login_required
 from django.contrib.auth.decorators import user_passes_test
 from django.db.models import Q
-from django.views.generic import ListView, DetailView, CreateView, UpdateView, TemplateView
+from django.views.generic import ListView, DetailView, CreateView, UpdateView, TemplateView, View
```
Comment (aedmondson) [inline ComSemApp/student/views.py:11]: View doesn't need to be imported

File: ComSemApp/student/views.py [hunk 2]
```diff
@@ -688,15 +688,31 @@ class SpeakingPracticeGeneratorView(StudentViewMixin, CourseViewMixin, TemplateV
     def foo():
         pass
 
-class SpeakingPracticeResultsView(StudentViewMixin, CourseViewMixin, TemplateView):
+practice_data = [
```
Comment (aedmondson) [inline ComSemApp/student/views.py:691]: Mocking this data is good for now. We will change this later

File: ComSemApp/student/views.py [hunk 3]
```diff
@@ -688,15 +688,31 @@ class SpeakingPracticeGeneratorView(StudentViewMixin, CourseViewMixin, TemplateV
...
     template_name: str = 'ComSemApp/student/assessment_results.html'
 
-    def foo():
-        pass
+    def get_object(self):
+        return self.course
+
+
+    def get_context_data(self, **kwargs):
+        """Get worksheet data for a student in a given course
```
Comment (aedmondson) [inline ComSemApp/student/views.py:707]: This docstring needs changed

File: ComSemApp/templates/ComSemApp/student/assessment_results.html [hunk 4]
```diff
@@ -1,5 +1,72 @@
...
+	<div class="col-12 h-100">
+        <!-- invisible until setup is complete -->
+		<div class="card-box">
+			<h4 class="header-title m-t-0 m-b-20">Verbal Practice</h4>
+			<!-- bar displaying student's progress throughout the reviewsheet -->
+			<div class="progress"  style="height: 20px;">
+  				<div class="progress-bar bg-success" role="progressbar" style="height: 20px; width: 100%" aria-valuenow="100" aria-valuemin="100" aria-valuemax="100"></div>
+			</div>
+			<div class="card">
+                <div class='card-body'> <!--card for displaying results-->
```
Comment (aedmondson) [inline ComSemApp/templates/ComSemApp/student/assessment_results.html:16]: Move comments to line above the line(s) they describe

File: ComSemApp/templates/ComSemApp/student/assessment_results.html [hunk 5]
```diff
@@ -1,5 +1,72 @@
...
+                                <th class="text-center">Accuracy Score</th> <!--displays if term is gramatically correct-->
+                                <th class="text-center">Fluency Score</th> <!--displays if the student's answer was correct-->
+								<th class="text-center invisible"><button class="show-more" type="button">Show More</button></th>
+							</tr>
+                        </thead>
+                        <tbody id="result_body">
+							{% for item in practice_data%}
+							<tr>
+									<tr>
+										<td>
```
Comment (aedmondson) [inline ComSemApp/templates/ComSemApp/student/assessment_results.html:31]: The indentation in here needs to be made more consistent.

File: ComSemApp/templates/ComSemApp/student/assessment_results.html [hunk 6]
```diff
@@ -1,5 +1,72 @@
...
+										<td class="text-center">
+											{{item.fluency1}}%
+										</td>
+										<td class="text-center">
+											<button class="show-more" type="button" url=>Show More</button>
+										</td>
+						
+							{% endfor %}
+							<script>
+								$(function(){
```
Comment (aedmondson) [inline ComSemApp/templates/ComSemApp/student/assessment_results.html:46]: Anonymous functions should be declared as arrow functions () => {}

File: ComSemApp/templates/ComSemApp/student/assessment_results.html [hunk 7]
```diff
@@ -1,5 +1,72 @@
...
+									// bind edit and delete buttons to their functions
+									$('.show-more').on('click', function(e){
```
Comment (aedmondson) [inline ComSemApp/templates/ComSemApp/student/assessment_results.html:48]: Change the anonymous function to an arrow function (e) =>

**Review by aedmondson — CHANGES_REQUESTED** [review]

File: ComSemApp/templates/ComSemApp/student/assessment_results.html [hunk 8]
```diff
@@ -1,5 +1,72 @@
...
+			<div class="card">
+                <div class='card-body'> <!--card for displaying results-->
+                    <h1 class="card-title">Review complete!</h1>
+                    <table id="results_table" class="table table-hover"> <!--table for displaying results of a worksheet-->
+                        <thead>
+                            <tr>
+                                <th>Student Transcription</th> <!--display term that students saw in question-->
+                                <th class="text-center">Accuracy Score</th> <!--displays if term is gramatically correct-->
+                                <th class="text-center">Fluency Score</th> <!--displays if the student's answer was correct-->
+								<th class="text-center invisible"><button class="show-more" type="button">Show More</button></th>
```
Comment (aedmondson) [inline ComSemApp/templates/ComSemApp/student/assessment_results.html:24]: This button class should also include a button style from [Bootstrap](https://getbootstrap.com/docs/4.0/components/buttons/)

**Review by Jared-Torii — APPROVED** [review]

**Review by aedmondson — APPROVED** [review]

**Pull request merged**